SimpleBrowser/Browser.cs

```diff
@@ -1281,11 +1289,15 @@ private IHttpWebRequest PrepareRequestObject(Uri url, string method, string cont
 		private void Register(Browser browser)
 		{
 			_allWindows.Add(browser);
+            if(!_allContexts.Contains(_allWindows)){
+                _allContexts.Add(_allWindows);
+            }
 			if (browser.WindowHandle == null)
 			{
 				browser.WindowHandle = Guid.NewGuid().ToString().Substring(0, 8);
 			}
 		}
+        private static List<List<Browser>> _allContexts = new List<List<Browser>>();
 

```

Comment:
What is the semantic meaning of the possibility of access to all your open pages of all application threads working with this class?